pennylane/gradients/hadamard_gradient.py

```diff
@@ -270,22 +276,43 @@ def hadamard_grad(
         Array([-0.3875172 , -0.18884787, -0.38355704], dtype=float64)
     """
 
-    transform_name = "Hadamard test"
+    modes = {
+        "standard": ("Hadamard test", _hadamard_test),
+        "reversed": ("Reversed hadamard test", _reversed_hadamard_test),
+        "direct": ("Direct hadamard test", _direct_hadamard_test),
+        "reversed-direct": ("Reversed direct hadamard test", _reversed_direct_hadamard_test),
+    }
+    try:
+        transform_name, gradient_method = modes[mode]
+    except KeyError as exc:
+        raise ValueError(f"Invalid mode: {mode}") from exc
```

Comment:
I would suggest the separate the ALLOWED_MODE_DICT out from the local scope of our `hadamard_grad` (or even consider `Enum`ize them).

Reply:
Also, it will be better if we can directly check existence of keyvalue to raise the same ValueError instead of a single try-except structure here.

Reply:
A simple test for non-existing keyvalue is required as well.

Reply:
Generally we don't want to use errors as logic, but here we are just adding information and switching the error type.  So I'm fine with using a try-except. 

But yes, will need a test. I can push up a blank test for this case.